hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1732,6 +1760,45 @@ int MDMParser::_cbCNUM(int type, const char* buf, int len, char* num)
     return WAIT;
 }
 
+int MDMParser::_cbUCGED(int type, const char* buf, int len, NetStatus* status)
+{
+    if ((type == TYPE_PLUS || type == TYPE_UNKNOWN) && status) {
+        int rsrp;
+        int rsrq_n;
+        unsigned long rsrq_f;
+        // AT+UCGED?
+        // +RSRP: 314,5110,"-102.60",163,5110,"-097.90",173,5110,"-100.40",
+        // +RSRQ: 314,5110,"-19.60",163,5110,"-16.60",173,5110,"-18.60",
+
+        // RSRP maps from dBm [-140,-44] to [0,97]
+        if (sscanf(buf, "\r\n+RSRP: %*d,%*d,\"%d\"",&rsrp) == 1){
```

Comment:
When I added `%*lu` for the fractional part, the compiler is complaining, so I am doing `%u`? :)

`use of assignment suppression and length modifier together in gnu_scanf format`

Reply:
The compiler is right :) Since we are suppressing assignment, `scanf` doesn't really care about the range/size of the argument variable and only about what characters may represent it.